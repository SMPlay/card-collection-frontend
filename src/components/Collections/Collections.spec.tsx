import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { create } from "react-test-renderer";
import { mount } from "enzyme";

import { Collections, CollectionsProps } from "./collections";

describe("Collections component", () => {
  let props: CollectionsProps;
  let component;
  beforeEach(() => {
    props = {
      collections: [
        {
          id: "test",
          name: "test",
          imageUrl: "test",
          cardsCount: 1,
          release: 1,
        },
      ],
      loading: false,
      error: false,
    };
  });

  it("should render", () => {
    const tree = create(
      <Router>
        <Collections
          collections={props.collections}
          loading={props.loading}
          error={props.error}
        />
      </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should render with loading", () => {
    component = mount(
      <Router>
        <Collections
          collections={props.collections}
          loading={true}
          error={props.error}
        />
      </Router>
    );

    expect(component.exists("Loading")).toBeTruthy();
  });

  it("should render with error", () => {
    component = mount(
      <Router>
        <Collections
          collections={props.collections}
          loading={props.loading}
          error={true}
        />
      </Router>
    );

    expect(component.exists("ReceivedError")).toBeTruthy();
  })
});
