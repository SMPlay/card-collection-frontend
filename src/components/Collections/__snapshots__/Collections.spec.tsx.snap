// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collections component should render 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
  >
    <a
      href="/collections/test"
      onClick={[Function]}
    >
      <div
        className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <img
            alt="test"
            className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
            height="400"
            src="test"
            title="test"
          />
          <div
            className="MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
            >
              test
            </h2>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiCardActionArea-root makeStyles-metaText-2"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <h5
            className="MuiTypography-root MuiTypography-h6"
          >
            Колличество карт: 
            1
          </h5>
          <h5
            className="MuiTypography-root MuiTypography-h6"
          >
            Год выпуска: 
            1
          </h5>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
        </button>
      </div>
    </a>
  </div>
</div>
`;
